Swift 2 is here!

saghul opened the pull request:
Ahoi!

Thanks for making this plugin available! :heart: 

Swift 2.0 was released yesterday as part of Xcode 7.0, so it's time to upgrade! With the attached patches I could compile the iOSAppRTC example using Xcode 7 and deploy it to an iPad running iOS 9.

IMPORTANT NOTE: As of right now Apple requires applications to include Bitcode, but it has to be disabled for now since the bundled libwebrtc was built without it. I left a note about this on the documentation.

saghul commented:
/cc @ibc

ibc commented:
Wow, amazing!

Let me please merge it tomorrow (must update Xcode and so on).

joshuairl commented:
Very anxious!  Thank you @saghul for spearheading this initiative.
And thank you as always @ibc for all of your hard work and commitment to making all of this possible.

windht commented:
Thank you @saghul !

saghul commented:
Updated hook to do the bitcode dance for us. (I guess iOSRTCApp would need updating after this lands)

joshuairl commented:
Forgive my ignorance to this process but does updating this to Swift 2.0 cause all people installing the app to be forced to install iOS 9 before they can use my app?

saghul commented:
@joshuairl I don't think so. I happen to have iOS 9 on my test device, but the deployment target was still set to 8.2. Alas, I have no iOS 8 device to confirm this.

windht commented:
@joshuairl 
I have made a dirty version to get swift 2 work a month ago because I updated to el capitan and had to use xcode 7 beta, but my iphone remains ios 8.4 and everything works pretty well. So no worries.

joshuairl commented:
@windht @saghul could one of you help me understand this... so I decided to try the swift-2 branch of @saghul's PR repo and so I removed iosrtc plugin from cordova.

and did a `cordova plugin add https://github.com/saghul/cordova-plugin-iosrtc#swift-2` to install it.
and it seemed to install correctly...

I'm now getting:
`ERROR: Plugin 'iosrtcPlugin' not found, or is not a CDVPlugin. Check your plugin mapping in config.xml.`

mgerstenblatt commented:
@joshuairl I don't know if this will help...but open up the project in xCode and expand the plugins folder  in the project navigator. Select 'iosrtcPlugin.swfit' and make sure that the checkbox under Target Membership (in the right window pane) is checked off. 

I often find these unchecked when removing/adding plugins without blowing away the entire platform.

saghul commented:
I fixed and force pushed a bit earlier today. Can you delete all the
downloaded files and retry?

On Friday, September 18, 2015, Joshua Rountree <notifications@github.com>
wrote:

> @windht <https://github.com/windht> @saghul <https://github.com/saghul>
> could one of you help me understand this... so I decided to try the swift-2
> branch of @saghul <https://github.com/saghul>'s PR repo and so I removed
> iosrtc plugin from cordova.
>
> and did a cordova plugin add
> https://github.com/saghul/cordova-plugin-iosrtc#swift-2 to install it.
> and it seemed to install correctly...
>
> I'm now getting:
> ERROR: Plugin 'iosrtcPlugin' not found, or is not a CDVPlugin. Check your
> plugin mapping in config.xml.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eface2face/cordova-plugin-iosrtc/pull/70#issuecomment-141535909>
> .
>


-- 
/Saúl
bettercallsaghul.com

joshuairl commented:
Seems to be working okay as for now! :)

joshuairl commented:
Somethings not quite right.
Thinking the select audio output function isn't working right?

saghul commented:
@joshuairl I haven't tried that. Can you give some more details on what's not going right? Any log lines?

joshuairl commented:
Honestly haven't even checked logs because my damn device updated to 9.1 and can't get Xcode to install haha.

However I'm sure it's unrelated but my initial stream won't record on MCU.  Again, I'm sure it's unrelated but who knows haha.

I'm pretty new to webrtc.

joshuairl commented:
Okay, figured out my MCU issue... obviously that wasn't even related.
I'm about to hook up another iOS device that is running iOS 9.0 to check logs.

I continue to have seemingly intermittent issues with where one side cannot hear anything from other side... although the sound is transmitting because my MCU is capturing it so I know the transmitting is working... but thinking either the "audio output device" isn't firing or something...

joshuairl commented:
Going to try to make sure I have pulled the latest code from your repo to ensure I'm not running outdated stuff.

ibc commented:
@joshuairl this plugin uses the Google libwebrtc library as it is. It does not handle audio/video by itseld in any way. If you experiment issues with your MCU please check the same environment with your WebRTC browser application, but please let's not comment these issues in this ticket about migrating to Swift 2.0.

joshuairl commented:
So some how I was able to continue running the old Swift code and compile just fine now... I just have to hit "Cancel" when it asks me to convert the code... before it would go a head and screw up all of the code references and wouldn't let me compile it.

But now I am experiencing a couple of intermittent peer connections not displaying correctly.
Still trying to get my iPhone back to 9.0 to run it live on the device.  Grrr :p

@ibc Sorry about that, I recognize that it isn't an iosrtc issue.  It turns out that it was something I did (of course!) :)

I just know how important it is to always include details into issues that may not always present obvious side-effects.  Hoping to just make the project as solid as possible!

Thanks for all you guys do.

mgerstenblatt commented:
@ibc any plans on merging this into master soon?

saghul commented:
@mgerstenblatt Hey, @ibc is afk for a few days. If this PR working fine for you?

joshuairl commented:
Yeah I'm just having a lot of trouble getting `cordova.plugins.iosrtc.selectAudioOutput('speaker')` to work.
I feel like it was working before but who knows.

Anybody else having any issues?

ibc commented:
@joshuairl `selectAudioOutput()` did work before, but remember it only works if you call it **once** a WebRTC session is started. Anyhow, if you detect it fails under those conditions, please open a new issue describing it.

psirenny commented:
I get [these errors](https://gist.github.com/psirenny/4513943d33d2d7472876) when I try to build. I'm running Xcode 7.1 on El Capitan. I am running the swift hook in the extras folder.

saghul commented:
@psirenny Looks like you haven't applied the changes from this PR. Those `.value` become `.rawValue`, for example.

ibc commented:
@saghul I'm back. I've created a new branch "swift1.2" to hold the current Swift 1.2 based plugin, and will merge your PR on master right now if you confirm me that's ready :)

ibc commented:
Guys, cordova-plugin-iosrtc 2.0.0 is here. It requires Swift 2.0.0 so thanks a lot to @saghul for the PR :cake:

saghul commented:
\o/
On Oct 4, 2015 15:42, "Iñaki Baz Castillo" <notifications@github.com> wrote:

> Guys, cordova-plugin-iosrtc 2.0.0 is here. It requires Swift 2.0.0 so
> thanks a lot to @saghul <https://github.com/saghul> for the PR [image:
> :cake:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eface2face/cordova-plugin-iosrtc/pull/70#issuecomment-145350894>
> .
>
